### ISSUE TITLE
Executing EXE file with arguments

### ISSUE THREAD

**Issue by hegusung** [fancycode/MemoryModule#36]
Hi,
first of all great job, this module works really well with all the DLL file I created.

However I know want to execute EXE files with arguments from memory, It seems your code is also able to execute EXE files, but I don't know were I should provide the arguments.

Could you tell me the right way to do this ?

**Comment by glysbaysb** [comment]
Well how does your executable get its arguments?

Is it via GetCommandLineA()/GetCommandLineW()? You can either hook that or replace the saved pointer.
`MOV EAX, DS:[some address]
 RETN `
Starting with Win 7 (IIRC) it's in another DLL so you have to search in kernelbase.dll

Is it via `int main(int argc, char** argv)`? That is set by your CRT, the one provided by Visual Studio uses msvcrt.dll's __getmainargs` which uses `_acmdln` (Both symbols are exported). You can either hook that or replace the saved pointer.

**Comment by hegusung** [comment]
I don't want the solution to be limited to only one executable, I would like it to be working as if I did : 
executables.exe -arg1 -arg2

is there a way to do that with MemoryModule ?

**Comment by vyrus001** [comment]
i have added support for arguments in a pull request (that has since been accepted) to a fork by another person -> https://github.com/kost/MemoryModule once his changes are included, arguments will work

**Comment by FusixGit** [comment]
I have made myself in the form of something like.
https://github.com/FusixGit/MemoryModule

**Comment by vyrus001** [comment]
This can be done with MemoryLoadLibraryEx() by setting the *userdata to *argv[]. Both of our patches are not needed. (I recommend this issue be closed)

**Comment by viaMorgoth** [comment]
If I set `*userdata` to a custom array but do not create a new process, the child executable seems to take the command line arguments of the parent process, and _not_  my array. Is there a way to do this without spawning a new process, or forcing the command line args passed into the parent process?

**Comment by kernel-sanders** [comment]
@vyrus001 Not quite. The exe will still call `GetCommandLine` which returns whatever args were passed to the programing using MemoryModule. In order to pass args to a memory loaded exe you will have to find and replace the `GetCommandLine` function with a pointer to a string in memory that contains the arguments.

**Comment by noname29** [comment]
@kernel-sanders any illustration for that ?

**Comment by kernel-sanders** [comment]
It has been done in other projects, see: [Invoke-ReflectivePEInjection.ps1#L2058](https://github.com/clymb3r/PowerShell/blob/master/Invoke-ReflectivePEInjection/Invoke-ReflectivePEInjection.ps1#L2058)

However, it is likely easier to just use [donut](https://github.com/TheWover/donut).

**Comment by noname29** [comment]
how to pass arguments  to a dll ? @kernel-sanders

**Comment by kernel-sanders** [comment]
It looks like you are attempting to write cryptomining malware. I suggest spending your talents elsewhere.

**Comment by noname29** [comment]
not really. I am making experiments. @kernel-sanders i work with security

**Comment by tehtris4** [comment]
![not really](https://i.imgflip.com/4wh8jx.jpg)

**Comment by n1nj4sec** [comment]
![not really](https://i.imgflip.com/4wh8jx.jpg)
